### ISSUE TITLE
ld: symbol(s) not found for architecture x86_64 clang: error: linker command failed with exit code 1

### ISSUE THREAD

**Issue by Zelfapp** [flutter/flutter#88641]
## Steps to Reproduce

Trying to run app (that compiles and works perfectly on android) on ios.

@jmagman requested I create a new issue after switching to `flutter beta` channel to try to take advantage of #85059 and #87244. See #87908, my original issue on `flutter stable`.

It still will not build on beta. Verbose logs and other info attached in large zips. 

Trying to run in devtools in Android Studio here's a summary of what is in the attached verbose log of what fails now on beta:

<details>
<summary>error attempting to run in AS on ios simulator</summary>

```
The plugins `cloud_functions, flutter_contact, flutter_uploader` use a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if these plugins support the Android V2 embedding. Otherwise, consider removing them since a future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to the V2 embedding: https://flutter.dev/go/android-plugin-migration.
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running pod install...
Running Xcode build...
Xcode build done.                                           14.0s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Command CompileSwift failed with a nonzero exit code
    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_GULAppEnvironmentUtil", referenced from:
          objc-class-ref in FIRCLSApplication.o
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
```
</details>

## Logs

@jmagman as requested here are the `--verbose` logs on `flutter beta` channel.
[flutter-run-verbose-beta-channel.zip](https://github.com/flutter/flutter/files/7025536/flutter-run-verbose-beta-channel.zip)

Also, included all the items @darshankawar had requested at some point.
[ios-xcode-build-files.zip](https://github.com/flutter/flutter/files/7025537/ios-xcode-build-files.zip)

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 2.5.0-5.2.pre, on macOS 11.5.1 20G80 darwin-x64, locale en-US)
    • Flutter version 2.5.0-5.2.pre at /Users/nate/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 19c61fed0d (3 days ago), 2021-08-18 17:10:31 -0700
    • Engine revision 7a4c4505f6
    • Dart version 2.14.0 (build 2.14.0-377.7.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/nate/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.2

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Connected device (1 available)
    • iPhone 12 Pro Max (mobile) • C90B29AC-5AD4-4BC2-A379-EAA45B77D5ED • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)

• No issues found!
```
</details>

**Comment by darshankawar** [comment]
@Zelfapp 
The new issue request was for the other commenter who mentioned that they are facing same issue using another package, https://github.com/flutter/flutter/issues/87908#issuecomment-898326763

For you, the request was to switch to latest beta and try to see if same issue occurs there as well, https://github.com/flutter/flutter/issues/87908#issuecomment-902922284

So instead of creating new issue for same reason, please share your latest beta findings in the original issue https://github.com/flutter/flutter/issues/87908

Closing from here as a self-duplicate.

**Comment by jmagman** [comment]
@Zelfapp Sorry for the confusion!
